static/js/controllers/contacts-edit.js

```diff
@@ -34,7 +34,7 @@ var _ = require('underscore');
         return UserDistrict()
           .then(function(facilityId) {
             if (!facilityId) {
-              return $q.resolve();
+              return $q.reject('Facility ID not supplied');
```

Comment:
Ah. Then that's a problem indeed. 
I haven't tested it, just read code and thought this would do the trick : 
https://github.com/medic/medic-webapp/blob/develop/static/js/services/user.js#L43